api.php

```diff
@@ -0,0 +1,211 @@
+	$user = User::findUserById(intval($id));
+	$body = json_decode($request->getBody());
+
+	//Update user info
+	if (isset($body->first_name)) {
+		$user->setFirstName($body->first_name);
+	}
+
+	if (isset($body->last_name)) {
+		$user->setLastName($body->last_name);
+	}
+
+	if (isset($body->email)) {
+		$user->setEmail($body->email);
+	}
+
+	if (isset($body->password)) {
+		$user->setPassword($body->password);
+	}
+
+	if (isset($body->phone)) {
+		$user->setPhoneNumber($body->phone);
+	}
+	var_dump($user);
+
+	$response->getBody()->write(json_encode($user));
+	return $response
+		->withHeader('Content-Type', 'application/json')
+		->withStatus(201);
+});
+
+
+/********* REVIEW ROUTES *********/
+
+// /reviews - displays all reviews
+$app->get('/reviews', function (Request $request, Response $response, array $args) {
+	// get our sort/filter vals
+	$params = $request->getQueryParams();
+	$filterVal = (isset($params['filter-by']) ? $params['filter-by'].' '.$params['filter-val'] : '');
+	$reviews = Review::getAllReviews($filterVal, $params['archived'] ?? false);
+	//return review info based on ID
+	$response->getBody()->write(json_encode($reviews));
+	return $response
+		->withHeader('Content-Type', 'application/json')
+		->withStatus(200);
+});
+
+// /review/{reviewID} - displays the info about a specific review not just the review contents
+$app->get('/review/{reviewID}', function (Request $request, Response $response, array $args) {
+	$reviewID = intval($args['reviewID']);
+	$review = \Review::getReviewByID($reviewID);
+	//return review info based on ID
+	$response->getBody()->write(json_encode($review));
+	return $response
+		->withHeader('Content-Type', 'application/json')
+		->withStatus(200);
+});
+
+//review/new  Creates a new review
+$app->post('/review/new', function (Request $request, Response $response, array $args) {
+	$body = json_decode($request->getBody());
+
+	if (empty($body->score) || empty($body->comment)) {
+		return $response->withStatus(400);
+	}
+
+	if (empty($body->userID) || intval($body->userID) == 0) {
+		return $response->withStatus(400);
+	}
+
+	if (!intval($body->score) > 0) {
+		return $response->withStatus(400);
+	}
+
+	$review = new Review($body->score, $body->comment, $body->userID);
+	if ($review->saveToDB()) {
+		$response->getBody()->write(json_encode($review));
+		return $response
+			->withHeader('Content-Type', 'application/json')
+			->withStatus(201);
+	}
+
+	return $response->withStatus(500);
+});
+
+//review/{reviewid} Update a review
+$app->post('/review/{reviewId}', function (Request $request, Response $response, array $args) {
+	$id = $args['reviewId'];
+	$body = json_decode($request->getBody());
+	//check that the review has an id and user id
+	if (intval($id)) {
+		$review = Review::getReviewByID($id);
+		//check that the score and comment are not empty
+		if (empty($body->score) && empty($body->comment)) {
+			return $response->withStatus(400);
+		}
+
+		//check to make sure score is more than 0
```

Comment:
this is accepting non integers as valid scores.

Reply:
fixed